docs/pages/tutorials/minimal/use-onchain.mdx

```diff
@@ -0,0 +1,328 @@
+- `script/ScriptClient.s.sol:ScriptClient` is the contract to run as a script.
+  By convention, forge scripts are stored in `script` and have the extension `.s.sol`.
+
+<details>
+
+<summary>Content of `script/ScriptClient.s.sol`</summary>
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+```
+
+This is standard Solidity boilerplate.
+
+```solidity
+import { Script } from "forge-std/Script.sol";
+import { console } from "forge-std/console.sol";
+```
+
+These lines are standard Foundry imports:
+
+- `lib/forge-std/Script.sol` is the script definitions.
+- `lib/forge-std/console.log` lets us emit console messages (you can see them in the output of the command above).
+
+```solidity
+import { Counter } from "codegen/Tables.sol";
+import { IWorld } from "codegen/world/IWorld.sol";
+```
+
+These lines import from the generated MUD code the definitions we need:
+
+- `Counter`, the counter with which we wish to interact.
+- `IWorld`, an interface to the `World` with all the functions from the various systems.
+
+```solidity
+contract ScriptClient is Script {
+  IWorld world;
+
+  function run() external {
+```
+
+This is the function that runs the script.
+
+```solidity
+    // Load the private key from the `PRIVATE_KEY` environment variable (in .env)
+    uint256 myPrivateKey = vm.envUint("PRIVATE_KEY");
+
+    // Start broadcasting transactions from that account
+    vm.startBroadcast(myPrivateKey);
+```
+
+Read the private key from the environment (which includes the content of the `.env` file) and use it to issue transactions.
+
+```solidity
+    // Connect to the world using `WORLD_ADDR` from .env
+    console.log("Connecting to world at ", vm.envAddress("WORLD_ADDR"));
+    world = IWorld(vm.envAddress("WORLD_ADDR"));
+```
+
+Connect to the `World` at `WORLD_ADDR`.
+
+```solidity
+    console.log("The counter values is now ", Counter.get(world));
+
+    console.log("Incrementing the counter");
+    world.increment();
+
+    console.log("The counter values is now ", Counter.get(world));
+```
+
+This is the actual script.
+Note that it has access to `Counter.get(world)` to read the data, but to modify it you have to go through a `System`.
+Remember, _there are no secrets on the blockchain_ and therefore we can assume that anybody can read anything at any time after it is posted - so we might as well provide read access directly.
+
+<details>
+
+<summary>Try to bypass security</summary>
+
+There is another script `BypassSecurity`, which attempts to bypass the `System` logic and set `Counter` directly.
+You can run it with this command.
+
+```sh copy
+forge script --rpc-url http://localhost:8545 --broadcast script/ScriptClient.s.sol:BypassSecurity
+```
+
+It fails with this error:
+
+```
+== Logs ==
+  Connecting to world at  0x5FbDB2315678afecb367f032d93F642f64180aa3
+  Just before trying to reset the counter
+Error:
+e64c0348:ROOT_NAMESPACE/Counter
```

Comment:
The error here is `AccessDenied`. Sad that forge is not able to display that properly, but maybe we can mention it here?